### PR TITLE
 update #555 - add development setup page

### PR DESCRIPTION
**Pull request by Ulisseus**
This PR adds development setup page (and other related documents like node on chromebook, cli and git introductions). 
It also adds mdx support to linters so lint-staged works with new format. 

I changed a few minor things. Node setup mentions Windows Subsystem for Linux for Windows users now (I believe that's what @kondanna is using). Developments setup page has a link to GitPod. 

[Vercel](https://c0d3-app-ceeorc9oc-c0d3.vercel.app/docs/setup).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,8 +19,8 @@
     "vercel-build": "./scripts/prismaUtils.js baseline migrate && next build"
   },
   "lint-staged": {
-    "*.{js,jsx,ts,tsx}": "eslint --fix",
-    "*.{json,md,scss}": "prettier --write",
+    "*.{js,jsx,ts,tsx,mdx}": "eslint --fix",
+    "*.{json,md,scss,mdx}": "prettier --write",
```
Comment (ggwadera) [inline package.json:23]: `eslint --fix` already applies prettier style rules.

```suggestion
    "*.{js,jsx,ts,tsx,mdx}": "eslint --fix",
    "*.{json,md,scss}": "prettier --write",
```

File: pages/docs/setup.mdx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+Now that you have your code, you need to run it with a program called
+**`node.js`**.
+
+1. Download the latest stable version of [node.js](https://nodejs.org/en/), **a
+   program to run JavaScript**. If you're using Windows install
+   [Windows Subsystem for Linux](https://docs.microsoft.com/en-us/windows/wsl/install-win10)
+   first and follow instructions for Linux from now on.
+
+   1. For companies that use `node`, the best practice is to use
+      [nvm](`https://github.com/nvm-sh/nvm`) to install node.js.
```
Comment (ggwadera) [inline pages/docs/setup.mdx:46]: Broken link

```suggestion
      [nvm](https://github.com/nvm-sh/nvm) to install node.js.
```

File: pages/docs/git.mdx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+(This example is overly simplified)
+
+## Codebase
+
+Treat the curriculum like you have joined a new company and working on a new
+codebase. The first thing you should do is to download it onto your machine. To
+do this, you need to run `git clone url-to-codebase`. For C0D3 curriculum, the
+url is
+[https://git.c0d3.com/song/curriculum](https://git.c0d3.com/song/curriculum)
```
Comment (ggwadera) [inline pages/docs/git.mdx:32]: Wow this is very outdated haha. This should point to github.

```suggestion
[https://github.com/garageScript/curriculum](https://github.com/garageScript/curriculum)
```

File: pages/docs/submitting.mdx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+   - If you run into a `Missing write access ... EACCES` error, you need to run
+     npm with sudo:
+     - `sudo npm i -g c0d3`
+
+1. Install [git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+   and familiarize yourself with [common git commands](/docs/git)
+1. Use git to [clone the curriculum](https://github.com/garageScript/curriculum)
+   into your machine
+   - `git clone [https://github.com/garageScript/curriculum](https://github.com/garageScript/curriculum)`
```
Comment (ggwadera) [inline pages/docs/submitting.mdx:36]: broken command

```suggestion
   - `git clone https://github.com/garageScript/curriculum`
```

File: pages/docs/submitting.mdx [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+Branch names: Usually branch names are for your own eyes only, so you do not
+have create branches with meaningful names.
+
+_Note: Except for steps 3 (reading the challenge) and 4 (open and editing the
+file), all the commands should be run in the terminal_
+
+1. Make sure you are on the master branch: `git checkout master`
+2. Create a new branch from master (pro tip: `p1` for problem 1)
+   - `git checkout -b p1`
+3. Go to [c0d3.com](http://c0d3.com), click on Curriculum, select lesson 0, and
```
Comment (ggwadera) [inline pages/docs/submitting.mdx:74]: link to relative path or to `https://www.c0d3.com` to skip the redirection server.

```suggestion
3. Go to [c0d3.com](/), click on Curriculum, select lesson 0, and
```

File: pages/docs/submitting.mdx [hunk 6]
```diff
@@ -0,0 +1,172 @@
+import LessonLayout from '../../components/mdx/Layout'
+
+# Submitting Challenges
+
+Create an account on [c0d3.com/signup](http://c0d3.com/signup) if you don't
```
Comment (ggwadera) [inline pages/docs/submitting.mdx:5]: relative href

```suggestion
Create an account on [c0d3.com/signup](/signup) if you don't
```

File: pages/docs/submitting.mdx [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+## Getting Feedback
+
+As a software engineer, after submitting a feature you will wait for other
+software engineers to review your code.
+
+Similarly, after submitting your solution, your solution will be reviewed by
+other engineers to make sure you are following best practices in your code.
+
+You should be able to see your submissions when you go to
+[c0d3.com](http://c0d3.com), click on Curriculum, and select lesson 0.
```
Comment (ggwadera) [inline pages/docs/submitting.mdx:129]: link to curriculum here would be better, also some better wording.

```suggestion
You should be able to see your submissions when you go to the
[Curriculum](/curriculum) and selecting a lesson.
```

**Comment by Ulisseus** [comment]
Links should be fixed now. Also, I added submitting challenges section into development setup. 

[Vercel.](https://c0d3-app-ceeorc9oc-c0d3.vercel.app/docs/setup)

**Review by ggwadera — APPROVED** [review]

**Review by songz — APPROVED** [review]

**Review by hwong0305 — APPROVED** [review]

**Pull request merged**